.gitmodules

```diff
@@ -0,0 +1,16 @@
+[submodule "linux"]
+	path = linux
+	url = git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+	shallow = true
+[submodule "llvm"]
+	path = llvm
+	url = https://git.llvm.org/git/llvm.git/
+	shallow = true
+[submodule "clang"]
+	path = clang
```

Comment:
llvm/tools/clang wouldn't work?

Reply:
No.  Can't add a submodule in the path of another submodule in git.

Reply:
You can use LLVM_EXTERNAL_{CLANG,LLD,POLLY}_SOURCE_DIR (look on https://llvm.org/docs/CMake.html for more info). These variables allow you to point to various projects that aren't in the usual nested location.